vision_evaluation/evaluators.py

```diff
@@ -270,6 +270,66 @@ def _calculate(self, targets, predictions, average):
         return sm.average_precision_score(targets, predictions, average=average)
 
 
+class PerClassAccuracyEvaluator(Evaluator):
+    """
+    Per Class Accuracy for multiclass classification
+    """
+
+    def __init__(self):
+        super(PerClassAccuracyEvaluator, self).__init__()
+
+    def reset(self):
+        super(PerClassAccuracyEvaluator, self).reset()
+        self.confusion_matrix = 0
+
+    def add_predictions(self, predictions, targets):
+        """ Evaluate a batch of predictions.
+        Args:
+            predictions: the model output numpy array. Shape (N, num_class)
+            targets: the golden truths. Shape (N,)
+        """
+        assert len(predictions) == len(targets)
+        assert len(targets.shape) == 1
+
+        prediction_cls = np.argmax(predictions, axis=1)
+        self.confusion_matrix = np.add(self.confusion_matrix, sm.confusion_matrix(targets, prediction_cls))
+
+    def get_report(self, **kwargs):
+        """ Get per class accuracy report.
+        Args:
+            labels : the list of class names the same order as the targets label index
+        """
+        assert 'labels' in kwargs
+        normalized_cm = self.confusion_matrix .astype('float') / self.confusion_matrix.sum(axis=1)[:, np.newaxis]
+        per_class_accuracy = np.nan_to_num(normalized_cm.diagonal())  # avoid nan output
+
+        return {'per_tag_performance': {kwargs['labels'][i]: {'accuracy': per_class_accuracy[i]} for i in range(len(per_class_accuracy))}}
+
+
+class PerClassAveragePrecisionEvaluator(MemorizingEverythingEvaluator):
+    """
+    Average Precision evaluator for both multi-class and multi-label classification
+    """
+
+    def __init__(self):
+        super().__init__()
+
+    def _get_id(self):
+        return 'per_tag_performance'
```

Comment:
it is called "per_tag" here. the class name is PerClass. and the argument name is 'labels'. is there a reason to use different names?